index.js

```diff
@@ -94,17 +84,65 @@ function RedisClient(stream, options) {
     this.parser_module = null;
     this.selected_db = null; // Save the selected db here, used when reconnecting
     this.old_state = null;
-    this.options = options;
 
     this.install_stream_listeners();
     events.EventEmitter.call(this);
 }
 util.inherits(RedisClient, events.EventEmitter);
 
+RedisClient.prototype.create_stream = function() {
+    var options = this.options;
+    var connection_options = this.connection_options;
+    var tls_options = options.tls;
+    var stream;
+
+    if (!connection_options) {
+        connection_options = this.connection_options = {};
+        if (options.path) {
+            //unix
+            this.address = connection_options.path = options.path;
+        } else {
+            //tcp or tls
+            connection_options.port = options.port || default_port;
+            connection_options.host = options.host || default_host;
+            connection_options.family = (options.family === 'IPv6') ? 6 : 4;
+            if (tls_options) {
+                Object.keys(tls_options).forEach(function(opt) {
```

Comment:
You can use a `for ... in` loop here without checking hasOwnProperty, since the options got copied earlier and therefor won't contain anything that does not belong here.